File: ros2-py-outputs/agr/autonomous_ground_robot/collision_guard_node.py
"""CollisionGuard — node implementation. EDIT ME.

Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
(generation-gap pattern): this file is where demo/application logic lives.
The architectural wiring (endpoints, parameters, lifecycle transitions) is in
collision_guard_node_base.py, which is regenerated on every pipeline run.
"""
import rclpy

from .collision_guard_node_base import CollisionGuardBase


class CollisionGuard(CollisionGuardBase):
    """collision_guard application logic."""

    def handle_cmdVelIn(self, msg) -> None:
        """/cmd_vel_smooth (Twist).

        TODO: implement message processing logic.
        """

    def handle_scanSub(self, msg) -> None:
        """/scan (LaserScan).

        TODO: implement message processing logic.
        """


def main(args=None):
    rclpy.init(args=args)
    node = CollisionGuard()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ros2-cpp-outputs/agr/src/collision_guard_node.cpp
// CollisionGuard — node implementation. EDIT ME.
//
// Generated ONCE by the ros2-sysmlv2 bridge pipeline and never overwritten
// (generation-gap pattern): this file is where demo/application logic lives.
// The architectural wiring is in collision_guard_node_base.hpp/.cpp, regenerated
// on every pipeline run.
#include "autonomous_ground_robot/collision_guard_node_base.hpp"

class CollisionGuard : public CollisionGuardBase
{
public:
  using CollisionGuardBase::CollisionGuardBase;

protected:
  void handle_cmdVelIn(const geometry_msgs::msg::Twist & msg) override
  {
    // /cmd_vel_smooth — TODO: implement message processing logic.
    (void)msg;
  }
  void handle_scanSub(const sensor_msgs::msg::LaserScan & msg) override
  {
    // /scan — TODO: implement message processing logic.
    (void)msg;
  }
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<CollisionGuard>();
  rclcpp::spin(node->get_node_base_interface());
  rclcpp::shutdown();
  return 0;
}
